File: benchmark/c/cpw/find_number_perfect_squares_two_given_numbers/find_number_perfect_squares_two_given_numbers.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int find_number_perfect_squares_two_given_numbers ( int a, int b ) ;
int find_number_perfect_squares_two_given_numbers ( int a, int b ) {
  int cnt = 0;
  for ( int i = a;
  i <= b;
  i ++ ) for ( int j = 1;
  j * j <= i;
  j ++ ) if ( j * j == i ) cnt ++;
  return cnt;
}

File: translation_result/else_bms/fail/c/cpw/find_number_perfect_squares_two_given_numbers/gpt4/BaseRepair/find_number_perfect_squares_two_given_numbers.rs

fn find_number_perfect_squares_two_given_numbers(a: i32, b: i32) -> i32 {
    let mut cnt = 0;
    for i in a..=b {
        let mut j = 1;
        while j * j <= i {
            if j * j == i {
                cnt += 1;
            }
            j += 1;
        }
    }
    cnt
}
